src/olympia/reviewers/serializers.py

```diff
@@ -47,49 +47,34 @@ class Meta:
         )
 
 
-# NOTE: Because of caching, this serializer cannot be reused and must be
-# created for each file. It cannot be used with DRF's many=True option.
-class FileEntriesSerializer(serializers.ModelSerializer):
-    content = serializers.SerializerMethodField()
-    uses_unknown_minified_code = serializers.SerializerMethodField()
-    download_url = serializers.SerializerMethodField()
-    entries = serializers.SerializerMethodField()
-    selected_file = serializers.SerializerMethodField()
-    mimetype = serializers.SerializerMethodField()
-    sha256 = serializers.SerializerMethodField()
-    size = serializers.SerializerMethodField()
-    mime_category = serializers.SerializerMethodField()
-    filename = serializers.SerializerMethodField()
+class FileEntriesMixin(object):
+    def _get_version(self):
+        """If neither the current instance nor the parent instance is a Version,
+        get the cached version instance from the view."""
+        if isinstance(self.instance, Version):
+            return self.instance
 
-    class Meta:
-        fields = ('id', 'content', 'entries', 'selected_file', 'download_url',
-                  'uses_unknown_minified_code', 'mimetype', 'sha256', 'size',
-                  'mime_category', 'filename'
-                  )
-        model = File
+        if self.parent is not None and isinstance(
+                self.parent.instance, Version):
+            return self.parent.instance
+
+        return self.context['view'].get_version_object()
```

Comment:
When is this ever called in actual usage ? As I mentioned in my other comment The `get_version_object()` implementation in `ReviewAddonVersionMixin` never seems to be called by tests, but this is also true of this line - some serializer tests hit it, but not views tests, which is quite suspicious.

Reply:
I agree it seems suspicious. I'll have to look more carefully at the cases where it is called in serializer tests.

How are you able to tell when something is or isn't called via tests? Is there a coverage tool one can use when running tests to see which code is and is not covered?

Reply:
You can generate coverage with pytest (https://coverage.readthedocs.io/en/coverage-5.1/#quick-start - we disabled that a while back because it wasn't generating useful data for us, between how we split tests and how much legacy code there was with poor coverage) but in this case I simply added a breakpoint and ran the view tests, and it was never triggered.

Reply:
Ah, that makes sense!

Reply:
FWIW I just tested `coverage run -m pytest src/olympia/reviewers/tests/test_views.py -k ViewSet` and `coverage report -m src/olympia/reviewers/views.py` and it found the same thing - line 61 of `serializers.py` was missed, and so was line 1323 of `views.py`. I would have expected both to be hit when running those tests.

Reply:
Ok, I see the issue. This line is called in the tests when we are testing the `FileInfoSerializer` and the `FileInfoDiffSerializer` in isolation. In that case the serializer has an instance of a `File` object, not a `Version` object, and it doesn't have a `parent`. Hence it needs to call `self.context['view'].get_version_object()` to obtain a `Version` object.

In real life, those serializers are never called in isolation, but rather only as sub-serializers for the `file` property in their parent serializer. If we only need this code to support testing these in isolation, then we should remove it. I see two possibilities:
1. Write the tests differently so the serializer will have access to a `Version` object. I'm not sure this is actually practical, because these FileInfo... serializers are not designed to have access to a `Version` object, except via their parent. 
2. Remove the tests for the FileInfo... serializers in isolation and put those tests into the tests for the associated parent objects.

Number 2 sounds like the correct solution to me.

@diox WDYT?

Reply:
A possible alternative:
- Write the tests differently so that the serializer will have access to a `Version` via a parent. You'd need to set up the test in a certain way, but that should be doable.

Ultimately though, we should remove any dead code the views aren't actually using (Or, if it's not actually dead, write new views tests - normally I wouldn't necessarily test everything from the view perspective, but in this instance it's a fairly specific piece of view-serializer interaction so I think it would deserve it)

Reply:
This is all fixed now.